src/main/java/com/oauth/server/configuration/BabyActivitySkillsConfiguration.java

```diff
@@ -21,6 +23,6 @@
     @Bean
     public BabyActivityProfilePublisher babyActivityProfilePublisher() {
         return new BabyActivityProfilePublisher(tokenStore, partnerTokenManager,
-                new HttpClientFactory());
+                new HttpClientFactory(), Executors.newSingleThreadExecutor());
```

Comment:
SingleThreadExecutor will only have one thread in the pool. This will not work for concurrent requests. You can use Executors.newCachedThreadPool

Reply:
yup, fixed it